src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+                read(connFd, data, INSTANCE_DATA_LENGTH);
+                line = (data);
+
+                if (line.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
+                    result_wr = write(connFd, JasmineGraphInstanceProtocol::FILE_RECV_WAIT.c_str(),
+                                      JasmineGraphInstanceProtocol::FILE_RECV_WAIT.size());
+                    if (result_wr < 0) {
+                        instance_logger.error("Error writing to socket");
+                        *loop_exit_p = true;
+                        return;
+                    }
+                }
+            }
+            break;
         } else {
-            std::string createDirCommand = "mkdir -p " + aggregatorFilePath;
-            FILE *createDirInput = popen(createDirCommand.c_str(), "r");
-            pclose(createDirInput);
+            sleep(1);
+            continue;
         }
+    }
 
-        if (aggregatorHost == host) {
-            copyCommand = "cp " + centralStoreFile + " " + aggregatorFilePath;
-        } else {
-            copyCommand = "scp " + centralStoreFile + " " + aggregatorHost + ":" + aggregatorFilePath;
+    bzero(data, INSTANCE_DATA_LENGTH + 1);
+    read(connFd, data, INSTANCE_DATA_LENGTH);
+    line = (data);
+
+    if (line.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
+        instance_logger.log("Received : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
+        result_wr = write(connFd, JasmineGraphInstanceProtocol::FILE_ACK.c_str(),
+                          JasmineGraphInstanceProtocol::FILE_ACK.size());
+        if (result_wr < 0) {
+            instance_logger.error("Error writing to socket");
+            *loop_exit_p = true;
+            return;
         }
+        instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_ACK, "info");
+    }
 
-        FILE *copyInput = popen(copyCommand.c_str(), "r");
+    instance_logger.log("File received and saved to " + fullFilePath, "info");
+    *loop_exit_p = true;
 
-        if (copyInput) {
-            // read the input
-            while (!feof(copyInput)) {
-                if (fgets(buffer, 128, copyInput) != NULL) {
-                    result.append(buffer);
-                }
-            }
-            if (!result.empty()) {
-                std::cout << result << std::endl;
+    Utils::unzipFile(fullFilePath);
+    size_t lastindex = fileName.find_last_of(".");
+    string rawname = fileName.substr(0, lastindex);
+    fullFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + rawname;
+
+    while (!Utils::fileExists(fullFilePath)) {
+        bzero(data, INSTANCE_DATA_LENGTH + 1);
+        read(connFd, data, INSTANCE_DATA_LENGTH);
+        string response = (data);
+        response = Utils::trim_copy(response, " \f\n\r\t\v");
+        if (response.compare(JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK) == 0) {
+       
```

Comment:
Can we move `INSTANCE_DATA_LENGTH - 10` to a separate constant and refer that here? No need to define that in a separate constants file. Instead we can do that above this line.

Reply:
Ok